### PR TITLE
Show user's name (not email) in comments. #4522

### LINKED ISSUE #4522: Comments in session: show user's name (not email)
We should show the name of the user (or at least provide a way to retrieve the name) in comments. Currently, only the email is shown. The reader may not be able to connect the email to the name.

### PR DESCRIPTION
**Pull request by Mynk96**
Fixes #4522

### CODE REVIEW

**Comment by Mynk96** [comment]
UI changes
FeedbackResultsPage
session
StudentRecordsPage
records
InstructorSearchPage
s1

**Comment by Mynk96** [comment]
Ready for Review!

File: src/main/java/teammates/ui/pagedata/InstructorFeedbackResponseCommentAjaxPageData.java [hunk 1]
```diff
@@ -44,10 +47,9 @@ public FeedbackResponseCommentRow getComment() {
                 FeedbackParticipantType.STUDENTS,
                 FeedbackParticipantType.INSTRUCTORS
         };
-
         Map<FeedbackParticipantType, Boolean> responseVisibilities = new HashMap<>();
         for (FeedbackParticipantType type : relevantTypes) {
-            responseVisibilities.put(type, true);
+            responseVisibilities.put(type, question.isResponseVisibleTo(type));
```
Comment (Mynk96) [inline src/main/java/teammates/ui/pagedata/InstructorFeedbackResponseCommentAjaxPageData.java]: The above line fixes the problem stated in the [#comment](https://github.com/TEAMMATES/teammates/pull/7418#issuecomment-308003304) in PR 7418.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 2]
```diff
@@ -1760,7 +1760,6 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
         FeedbackSessionResponseStatus responseStatus = section == null && isIncludeResponseStatus
                                                      ? getFeedbackSessionResponseStatus(session, roster, allQuestions)
                                                      : null;
-
```
Comment (thenaesh) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: There is an extraneous change here.

File: src/main/java/teammates/ui/controller/InstructorFeedbackResponseCommentAddAction.java [hunk 3]
```diff
@@ -118,8 +118,9 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         data.commentId = commentId;
         data.showCommentToString = StringHelper.toString(createdComment.showCommentTo, ",");
         data.showGiverNameToString = StringHelper.toString(createdComment.showGiverNameTo, ",");
+        data.instructorEmailNameTable = bundle.instructorEmailNameTable;
+        data.question = logic.getFeedbackQuestion(feedbackQuestionId);
 
         return createShowPageResult(Const.ViewURIs.INSTRUCTOR_FEEDBACK_RESPONSE_COMMENTS_ADD, data);
     }
-
```
Comment (thenaesh) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResponseCommentAddAction.java]: And here.

File: src/main/java/teammates/ui/pagedata/InstructorFeedbackResponseCommentAjaxPageData.java [hunk 4]
```diff
@@ -44,10 +47,9 @@ public FeedbackResponseCommentRow getComment() {
                 FeedbackParticipantType.STUDENTS,
                 FeedbackParticipantType.INSTRUCTORS
         };
-
```
Comment (thenaesh) [inline src/main/java/teammates/ui/pagedata/InstructorFeedbackResponseCommentAjaxPageData.java]: Here too.

File: src/main/java/teammates/ui/pagedata/InstructorFeedbackResponseCommentAjaxPageData.java [hunk 5]
```diff
@@ -56,4 +58,5 @@ public FeedbackResponseCommentRow getComment() {
     public String[] getCommentIds() {
         return commentId.split("-");
     }
+
```
Comment (thenaesh) [inline src/main/java/teammates/ui/pagedata/InstructorFeedbackResponseCommentAjaxPageData.java]: Here.

File: src/main/java/teammates/storage/search/FeedbackResponseCommentSearchDocument.java [hunk 6]
```diff
@@ -318,8 +318,10 @@ public static FeedbackResponseCommentSearchResultBundle fromResults(
 
             String commentGiverName = extractContentFromQuotedString(
                     doc.getOnlyField(Const.SearchDocumentField.FEEDBACK_RESPONSE_COMMENT_GIVER_NAME).getText());
+
```
Comment (Mynk96) [inline src/main/java/teammates/storage/search/FeedbackResponseCommentSearchDocument.java]: @thenaesh Is this also a extraneous change?

File: src/main/java/teammates/ui/template/FeedbackResponseRow.java [hunk 7]
```diff
@@ -80,5 +82,4 @@ public String getResponseText() {
     public List<FeedbackResponseCommentRow> getResponseComments() {
         return responseComments;
     }
-
```
Comment (Mynk96) [inline src/main/java/teammates/ui/template/FeedbackResponseRow.java]: @thenaesh this one also?

**Comment by Mynk96** [comment]
@thenaesh Ready for Review!

**Comment by Mynk96** [comment]
@thenaesh Please Review!

**Pull request closed without merging**